### PR TITLE
#11 exception handling

### LINKED ISSUE #11: Create Global error handlers
### Problem Description
Spring Boot does not have a built-in NotFoundException or any similar kind; therefore, we need to create our own.

All exceptions need to be in the package `src/main/java/dev/ebrydeu/spring_boot_library/exception`.

https://www.baeldung.com/exception-handling-for-rest-with-spring

### PR DESCRIPTION
**Pull request by Charlottehenriksson**
### Related Issue(s):
Closes #11 

### Proposed Changes
Added exceptions handlers, is there any missing or something i need to add?

### Description
exception mapper added, and CustomExceptions class and ExceptionHandling class 

**- UserNotFoundException
- UserAlreadyExistsException
- EmailAlreadyExistsException
- InvalidCredentialsException
- AccessDeniedException
- DataIntegrityViolationException
- FileUploadException
- ValidationException
- ResourceNotFoundException
- InternalServerErrorException**

### CODE REVIEW

File: src/main/java/dev/ebrydeu/spring_boot_library/exception/CustomExceptions.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+package dev.ebrydeu.spring_boot_library.exception;
+
+public class CustomExceptions {
+    public static class UserNotFoundException extends RuntimeException {
+        public UserNotFoundException(String message) {
+            super(message);
+        }
+    }
```
Comment (Ebyrdeu) [inline src/main/java/dev/ebrydeu/spring_boot_library/exception/CustomExceptions.java:8]: Maybe make this more generic ? Like `NotFoundException`. The reason that if we have more than one Entity then each need own `XxxNotFoundException`.

**Review by Ebyrdeu — CHANGES_REQUESTED** [review]
Good job. Just one small thing need to be fixed and you're good.

File: src/main/java/dev/ebrydeu/spring_boot_library/exception/CustomExceptions.java (hunk 1 above)
Comment (Charlottehenriksson) [reply src/main/java/dev/ebrydeu/spring_boot_library/exception/CustomExceptions.java:8]: great I will fix this, thanks! :D

**Review by ajohnsson95 — APPROVED** [review]
Looks good :)

**Review by Ebyrdeu — APPROVED** [review]

**Pull request merged**